src/annotator/components/NotebookModal.tsx

```diff
@@ -40,9 +41,83 @@ function NotebookIframe({ config, groupId }: NotebookIframeProps) {
     />
   );
 }
+
+/**
+ * Checks if the browser supports native modal dialogs
+ */
+function isModalDialogSupported(document: Document) {
+  const dialog = document.createElement('dialog');
+  return typeof dialog.showModal === 'function';
+}
+
 export type NotebookModalProps = {
   eventBus: EventBus;
   config: NotebookConfig;
+
+  /** Test seam */
+  document_?: Document;
+};
+
+type WrapperProps = { isHidden: boolean; children: ComponentChildren };
+
+const DialogWrapper = ({ isHidden, children }: WrapperProps) => {
+  const dialogRef = useRef<HTMLDialogElement | null>(null);
+
+  useEffect(() => {
+    if (isHidden) {
+      dialogRef.current?.close();
+    } else {
+      dialogRef.current?.showModal();
+    }
+  }, [isHidden]);
+
+  // Prevent the dialog to be closed when `Esc` is pressed, to keep previous
+  // behavior
+  useEffect(() => {
+    const dialogElement = dialogRef.current;
+    const listener = (event: Event) => event.preventDefault();
+
+    dialogElement?.addEventListener('cancel', listener);
+
+    return () => {
+      dialogElement?.removeEventListener('cancel', listener);
+    };
+  }, []);
```

Comment:
>  I'm disabling this behavior for now, as in our case we need to send an event to the sidebar to open it again

Can you expand on this? Ideally we do want the dialog to respond to close requests in normal platform appropriate ways. The `cancel` event corresponds to the Escape key, but I think it can also correspond to other platform-specific triggers.

Reply:
> Ideally we do want the dialog to respond to close requests in normal platform appropriate ways

Yes, definitely. I will work on this after.

As of now, if we don't cancel this event, closing the modal via <kbd>Esc</kbd> key, does not bring the sidebar back, because it does not trigger the same logic the close button does.

I thought it would be easier to focus on that as an individual work unit, instead of bringing more stuff to this PR.